tools/reviewer/job/sync/DiffConverter.java

```diff
@@ -0,0 +1,173 @@
+import com.google.startupos.tools.reviewer.job.sync.GithubPullRequestProtos.ReviewComment;
+import com.google.startupos.tools.reviewer.job.sync.GithubPullRequestProtos.User;
+import com.google.startupos.tools.reviewer.localserver.service.Protos;
+import com.google.startupos.tools.reviewer.localserver.service.Protos.Comment;
+import com.google.startupos.tools.reviewer.localserver.service.Protos.Diff;
+import com.google.startupos.tools.reviewer.localserver.service.Protos.GithubPr;
+import com.google.startupos.tools.reviewer.localserver.service.Protos.Thread;
+
+import java.io.IOException;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+// Converts a Diff proto to a list of GithubPullRequestProtos.PullRequest protos
+public class DiffConverter {
+
+  private GithubClient githubClient;
+
+  public DiffConverter(GithubClient githubClient) {
+    this.githubClient = githubClient;
+  }
+
+  public List<PullRequest> toPullRequests(Diff diff) {
+    List<PullRequest> pullRequests = new ArrayList<>();
+
+    for (GithubPr githubPr : diff.getGithubPrList()) {
+      PullRequest.Builder pullRequest = PullRequest.newBuilder();
+
+      if (githubPr.getNumber() != 0) {
+        pullRequest.setNumber(githubPr.getNumber());
+      }
+      pullRequest
+          .setState(getPullRequestState(diff.getStatus()))
+          .setTitle("D" + diff.getId())
+          .setUser(User.newBuilder().setEmail(diff.getAuthor().getEmail()).build())
+          .setBody(diff.getDescription())
+          .setCreatedAt(Instant.ofEpochMilli(diff.getCreatedTimestamp()).toString())
+          .setUpdatedAt(Instant.ofEpochMilli(diff.getModifiedTimestamp()).toString())
+          .setBaseBranchName("master")
+          .setHeadBranchName("D" + diff.getId())
+          .setRepo(githubPr.getRepo())
+          .addAllReviewComment(getReviewCommentsByRepoName(diff, githubPr))
+          .addAllIssueComment(getIssueComments(diff, githubPr))
+          .setOwner(githubPr.getOwner())
+          .setAssociatedReviewerDiff(diff.getId());
+      pullRequests.add(pullRequest.build());
+    }
+    return pullRequests;
+  }
+
+  private List<ReviewComment> getReviewCommentsByRepoName(Diff diff, GithubPr githubPr) {
+    List<ReviewComment> result = new ArrayList<>();
+    List<Thread> codeThreads =
+        diff.getCodeThreadList()
+            .stream()
+            .filter(thread -> thread.getRepoId().equals(githubPr.getRepo()))
+            .collect(Collectors.toList());
+
+    for (Thread thread : codeThreads) {
+      for (Comment comment : thread.getCommentList()) {
+        ReviewComment.Builder githubComment = ReviewComment.newBuilder();
+        githubComment
+            .setPath(thread.getFile().getFilename())
+            .setId(comment.getGithubCommentId())
+            .setPosition(
+                thread.getGithubCommentPosition() == 0
+                    ? getReviewCommentPosition(thread, githubPr)
```

Comment:
Because `getReviewCommentPosition()` is a call to GitHub (heavy operation), it's better to:
* Have a separate variable for it, and do it in an `if-else`. That way it has "more space" and is more noticeable.
* Rename to something line `getReviewCommentPositionFromGitHub()` so it's obvious it's an external call.

Reply:
Done.